src/setup_payload/tests/run_python_setup_payload_test.py

```diff
@@ -0,0 +1,213 @@
+        'ProductID': None,
+        'Custom flow': None,
+        'Discovery Bitmask': None,
+        'Short discriminator': None,
+        'Long discriminator': None,
+        'Passcode': None
+    }
+
+
+def remove_escape_sequence(data):
+    ansi_escape = re.compile(r'\x1B(?:[@-Z\\-_]|\[[0-?]*[ -/]*[@-~])')
+    result = ansi_escape.sub('', data)
+    return result
+
+
+def chip_tool_parse_setup_payload(chip_tool, payload):
+    cmd_args = [chip_tool, 'payload', 'parse-setup-payload', payload]
+    data = subprocess.check_output(cmd_args).decode('utf-8')
+    data = remove_escape_sequence(data)
+    parsed_params = payload_param_dict()
+    for key in parsed_params:
+        k_st = data.find(key)
+        if k_st == -1:
+            continue
+
+        # 1 is for ":"
+        k_end = k_st + len(key) + 1
+
+        k_nl = data.find('\n', k_end)
+        parsed_params[key] = data[k_end:k_nl].split()[0]
+
+    return parsed_params
+
+
+def generate_payloads(in_params):
+    payloads = SetupPayload(in_params['Long discriminator'], in_params['Passcode'],
+                            in_params['Discovery Bitmask'], CommissioningFlow(in_params['Custom flow']),
+                            in_params['VendorID'], in_params['ProductID'])
+    manualcode = payloads.generate_manualcode()
+    qrcode = payloads.generate_qrcode()
+    return manualcode, qrcode
+
+
+def verify_generated_payloads(in_params, manualcode_params, qrcode_params):
+    assert in_params['Version'] == int(manualcode_params['Version'], 0)
+    assert in_params['Passcode'] == int(manualcode_params['Passcode'], 0)
+    assert in_params['Short discriminator'] == int(manualcode_params['Short discriminator'], 0)
+    if in_params['Custom flow'] != 0:
+        assert in_params['VendorID'] == int(manualcode_params['VendorID'], 0)
+        assert in_params['ProductID'] == int(manualcode_params['ProductID'], 0)
+
+    assert in_params['Version'] == int(qrcode_params['Version'], 0)
+    assert in_params['VendorID'] == int(qrcode_params['VendorID'], 0)
+    assert in_params['ProductID'] == int(qrcode_params['ProductID'], 0)
+    assert in_params['Custom flow'] == int(qrcode_params['Custom flow'], 0)
+    assert in_params['Discovery Bitmask'] == int(qrcode_params['Discovery Bitmask'], 0)
+    assert in_params['Passcode'] == int(qrcode_params['Passcode'], 0)
+    assert in_params['Long discriminator'] == int(qrcode_params['Long discriminator'], 0)
+
+
+def get_payload_params(discriminator, passcode, discovery=4, flow=0, vid=0, pid=0, version=0, short_discriminator=None):
+    p = payload_param_dict()
+    p['Version'] = version
+    p['VendorID'] = vid
+    p['ProductID'] = pid
+    p['Custom flow'] = flow
+    p['Discovery Bitmask'] = discovery
+    p['Long discriminator'] = discriminator
+    p['Short discriminator'] = short_discriminator if short_discriminator is not None else (discriminator >> 8)
+    p['Passcode'] = passcode
+    return p
+
+
+def test_code_generation(chip_tool):
```

Comment:
Suggest using plain old `unittest` module to convert the tests in here. Can be done as a follow-up.

Reply:
https://github.com/project-chip/connectedhomeip/issues/33549 for follow up